builder/builder.go

```diff
@@ -97,6 +98,9 @@ func New(dockerCli command.Cli, opts ...Option) (_ *Builder, err error) {
 			return nil, err
 		}
 	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
It's worth pointing out that it's possible to get no error but still have `b.NodeGroup == nil`:
https://github.com/milas/buildx/blob/e4a27ec8ab5a15ab12dcdf3bc4806f0852896eef/store/storeutil/storeutil.go#L63-L67

Let me know if you think we should return an error here in that case (`errors.New("no valid builder context")` or something) OR change `GetCurrentInstance` to propagate the error from the fallback call to `GetNodeGroup`

Reply:
I'm curious of the builder configuration. I guess there is an odd docker context being used as the default builder? Would be nice to have the output of `docker buildx ls` and `docker buildx inspect`.

Reply:
I don't quite understand this error check. How can it be nil at this point, without having already returned from the function?

Reply:
@crazy-max if there are no builders present, I think this is what causes that:

https://github.com/docker/buildx/blob/672eeed9a627d578746438e2bd611772ed7f9288/store/storeutil/storeutil.go#L54-L68

I think the proper fix here is to detect the error on the second lookup, and return that?

Reply:
Yep, there's two things:

The first, which is _currently_ covered by this PR is the missing error check after `NodeGroup` construction (i.e. either `GetNodeGroup` or `GetCurrentInstance`).

The second, linked a couple times in this thread ⬆️, which is NOT addressed by this PR [right now] is the lack of error propagation in `GetCurrentInstance`'s fallback call to `GetNodeGroup` - will not raise an error / segfault _here_ but will probably return a builder that is unusable and will 💥 on the next call

Given that e.g. `txn.Current()` can also return `nil, nil`, I think the best option is:
```go
	if b.opts.name != "" {
		if b.NodeGroup, err = storeutil.GetNodeGroup(b.opts.txn, dockerCli, b.opts.name); err != nil {
			return nil, err
		}
	} else {
		if b.NodeGroup, err = storeutil.GetCurrentInstance(b.opts.txn, dockerCli); err != nil {
			return nil, err
		}
	}
	if err != nil {
		return nil, err
	}

	// ⬇️  NEW ❗ 
	if b.NodeGroup == nil {
		return nil, errors.New("no builder available")
	}
	// ⬆️  NEW ❗ 


	if b.opts.validate {
		if err = b.Validate(); err != nil {
			return nil, err
		}
	}

	return b, nil
```

Reply:
(Could also do this in `Validate()` -- i.e. make that return an error if `b.NodeGroup == nil`)

Reply:
Looking at https://github.com/docker/compose/blob/af6f0ffb9e992502bad030d5c773f523ebd3ebc5/pkg/compose/build_buildkit.go#L36, we are creating a new Builder instance without specifying a name so it would get the current one: https://github.com/docker/buildx/blob/672eeed9a627d578746438e2bd611772ed7f9288/builder/builder.go#L95-L98

Having `NodeGroup` nil means there is no docker context which is not possible with Docker installed with what we know from https://github.com/docker/compose/issues/10453 imo.

I would have a preference for returning the error in https://github.com/docker/buildx/blob/672eeed9a627d578746438e2bd611772ed7f9288/store/storeutil/storeutil.go#L64

@tonistiigi Do you recall why we don't handle an error in this line?

Reply:
Just glancing over, but does the new error handling in this PR actually capture an error? Because both if/else branches before it seem to be returning an error (if any).

Reply:
Sorry for delay -- added error propagation from the fallback call based on context name as suggested above!